l10n_it_split_payment/models/account.py

```diff
@@ -100,37 +104,65 @@ def set_receivable_line_ids(self):
                         self.amount_total * line_client.credit
                     ) / inv_total
                 else:
-                    receivable_line_amount = 0
+                    receivable_line_amount = 0.0
                 line_client.with_context(check_move_validity=False).update(
                     {
                         "price_unit": -receivable_line_amount,
+                        "amount_currency": -receivable_line_amount,
+                        "debit": 0.0,
+                        "credit": receivable_line_amount,
                     }
                 )
 
     def _compute_split_payments(self):
         write_off_line_vals = self._build_debit_line()
         line_sp = self.line_ids.filtered(lambda l: l.is_split_payment)
         if line_sp:
-            if self.move_type == "out_invoice" and float_compare(
-                line_sp[0].debit,
-                write_off_line_vals["debit"],
-                precision_rounding=self.currency_id.rounding,
+            line_sp = line_sp[0].with_context(check_move_validity=False)
+            if (
+                self.move_type == "out_invoice"
+                and float_compare(
+                    line_sp[0].price_unit,
+                    write_off_line_vals["price_unit"],
+                    precision_rounding=self.currency_id.rounding,
+                )
+                != 0
             ):
-                line_sp[0].with_context(check_move_validity=False).update({"debit": 0})
+                line_sp.update(
+
+                    {
+                        "price_unit": 0.0,
+                        "amount_currency": 0.0,
+                        "debit": 0.0,
+                        "credit": 0.0,
+                    }
+                )
                 self.set_receivable_line_ids()
-                line_sp[0].debit = write_off_line_vals["debit"]
-            elif self.move_type == "out_refund" and float_compare(
-                line_sp[0].credit,
-                write_off_line_vals["credit"],
-                precision_rounding=self.currency_id.rounding,
+                line_sp[0].write(write_off_line_vals)
+            elif (
+                self.move_type == "out_refund"
+                and float_compare(
+                    line_sp[0].price_unit,
+                    write_off_line_vals["price_unit"],
+                    precision_rounding=self.currency_id.rounding,
+                )
+                != 0
             ):
-                line_sp[0].with_context(check_move_validity=False).update({"credit": 0})
+
+                line_sp.update(
+                    {
+                        "price_unit": 0.0,
+                        "amount_currency": 0.0,
+                        "debit": 0.0,
+                        "credit": 0.0,
+                    }
+                )
                 self.set_receivable_line_ids()
-                line_sp[0].credit = write_off_line_vals["credit"]
+                line_sp[0].write(write_off_line_vals)
```

Comment:
In questo punto userei direttamente: `line_sp.write(write_off_line_vals)` considerato che `line_sp = line_sp[0].with_context(check_move_validity=False)`. Puoi apportare la modifica? Poi ci dovremmo essere :+1: 
Grazie!